kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -562,7 +562,7 @@ private static Set<TableFeature> getDependencyFeatures(Set<TableFeature> feature
     }
   }
 
-  private static boolean hasInvariants(StructType tableSchema) {
+  public static boolean hasInvariants(StructType tableSchema) {
```

Comment:
For my erudition: Does making this `public` make it actually part of the kernel public api, or does the fact that this lives in `internal` mean its not actually exposed?

Reply:
Yeah so we have no concept of package private in Java 8... so anything accessible from another class needs to be `public`; but we put all of our internal code inside the `internal` package

Everything in the `internal` package is excluded from our java docs, but is still accessible publicly. There's nothing we can really do about that, and we aren't providing any guarantees around interfaces in the `internal` package.